recovery/recovery.go

```diff
@@ -0,0 +1,282 @@
+package recovery
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/orbs-network/scribe/log"
+)
+
+//const DDMMYYYYhhmmss = "2006-01-02-15:04:05"
+const (
+	e_zero_content   = "e_zero_content"
+	e_no_bash_prefix = "e_no_bash_prefix"
+	//e_content_not_changed = "e_content_not_changed"
+)
+
+/////////////////////////////////////////////////
+// JSON
+// {
+// 	"shell": {
+// 	"bin": "bash",
+// 	"run": [
+// 		"https://raw.githubusercontent.com/amihaz/staging-deployment/main/boyar_recovery/shared/disk_cleanup_1",
+// 		"https://raw.githubusercontent.com/amihaz/staging-deployment/main/boyar_recovery/shared/docker_cleanup_1"
+// 	]
+//   }
+// }
+type Shell struct {
+	Bin string   `json:"bin"`
+	Run []string `json:"run"`
+}
+
+type Instructions struct {
+	Shell Shell `json:"shell"`
+}
+
+type Config struct {
+	IntervalMinute uint
+	Url            string
+}
+
+type Recovery struct {
+	config      Config
+	ticker      *time.Ticker
+	tickCount   uint32
+	lastTick    time.Time
+	lastExec    time.Time
+	lastHash    string
+	lastOutput  string
+	lastReadErr string
+}
+
+var single *Recovery
+var logger log.Logger
+
+func Init(c Config, _logger log.Logger) {
+	//initialize static instance on load
+	logger = _logger
+	logger.Info("recovery - Init logger success")
+	single = &Recovery{config: c, tickCount: 0}
```

Comment:
I actually do want the singleton to be reconfigurable- in case for instance when boyar configuration changes and we want to change the url or the interval